### PR TITLE
[Documentation] dicomTar.pl

### PR DESCRIPTION
**Pull request by MounaSafiHarab**
Generated using:
`pod2markdown dicom-archive/dicomTar.pl > docs/scripts_md/dicomTar.md`

User can type:
perldoc dicom-archive/dicomTar.pl 
to get the documentation at the terminal

### CODE REVIEW

File: docs/scripts_md/dicomTar.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+Available options are:
+
+\-today                  : Use today's date for archive name instead of using
+                          acquisition date
+
+\-database               : Use a database if you have one set up for you.
+                          Just trying will fail miserably
+
+\-mri\_upload\_update      : update the mri\_upload table by inserting the correct
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: To be consistent with other descriptions, "update" should be written "Update the mri_upload table by inserting...)

File: docs/scripts_md/dicomTar.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+\-centerName             : Specify the symbolic center name to be stored
+                          alongside the DICOM institution
+
+\-verbose                : Be verbose if set
+
+\-version                : Print cvs version number and exit
+
+# DESCRIPTION
+
+A tool for archiving DICOM data. Point it to a source dir and provide a target
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: "dir" should be written "directory"

File: docs/scripts_md/dicomTar.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+dir which will be the archive location.
+
+\- If the source contains only one valid STUDY worth of DICOM it will create a
+  descriptive summary, a (gzipped) DICOM tarball
+  The tarball with the metadata and a logfile will then be retarred into the
+  final TARCHIVE.
+\- md5sums are reported for every step
+\- It can also be used with a MySQL database.
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: This list of items does not show as a list in the .md file and the perldoc.

File: docs/scripts_md/dicomTar.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+\-version                : Print cvs version number and exit
+
+# DESCRIPTION
+
+A tool for archiving DICOM data. Point it to a source dir and provide a target
+dir which will be the archive location.
+
+\- If the source contains only one valid STUDY worth of DICOM it will create a
+  descriptive summary, a (gzipped) DICOM tarball
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: need a "." after "DICOM tarball".

File: docs/scripts_md/dicomTar.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+\- md5sums are reported for every step
+\- It can also be used with a MySQL database.
+
+## Methods
+
+### archive\_head()
+
+Function that prints the tarchive header
+
+### read\_file()
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: Put the input variables in the name of the function to be consistent with other documentation.

File: docs/scripts_md/dicomTar.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+Function that reads file contents into a variable
+
+INPUT: $file : file to be read
+
+Returns: $content : file contents
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: "Returns" should be "RETURNS"

File: docs/scripts_md/dicomTar.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+None reported.
+
+# LICENSING
+
+License: GPLv3
+
+# AUTHORS
+
+J-Sebastian Muehlboeck
```
Comment (cmadjar) [inline docs/scripts_md/dicomTar.md]: Also add the LORIS team in the authors.

File: dicom-archive/dicomTar.pl [hunk 8]
```diff
@@ -42,27 +90,41 @@
...
+dir which will be the archive location.
+- If the source contains only one valid STUDY worth of DICOM it will create a
+  descriptive summary, a (gzipped) DICOM tarball
+  The tarball with the metadata and a logfile will then be retarred into the
+  final TARCHIVE.
 - md5sums are reported for every step
 - It can also be used with a MySQL database.
 
+
+Documentation: perldoc imaging_upload_file.pl
```
Comment (MounaSafiHarab) [inline dicom-archive/dicomTar.pl]: oopsie! copy and paste without thinking.

**Comment by MounaSafiHarab** [comment]
> Requesting a few minor changes in the documentation.
> 
> Tested insertion of DICOMs and all good!
> 
> Let me know once you have made the changes and I'll merge the PR.

I think I did all the requested changes! thanks a bunch...

File: dicom-archive/dicomTar.pl [hunk 9]
```diff
@@ -212,12 +313,13 @@
...
 
 exit 0;
-# **************************************************************************************************************************  
-=pod 
-################################################
-print tarchive header
-################################################
-=cut 
+# ******************************************************************************
+=head3 archive_head()
```
Comment (MounaSafiHarab) [inline dicom-archive/dicomTar.pl]: @cmadjar 
not sure if here I should have added `$finalTarget, $hostname, $system, ...)` to the list of inputs given that the function call does not pass them directly. What do you think? if yes, I can resend a separate pull request with only that change.

Comment (cmadjar) [reply dicom-archive/dicomTar.pl]: I would probably not add them here since they are not passed directly.

**Pull request merged**